src/pyuvdata/uvbeam/analytic_beam.py

```diff
@@ -0,0 +1,893 @@
+            polarization_array=polarization_array,
+            x_orientation=self.x_orientation,
+            pixel_coordinate_system=pixel_coordinate_system,
+            axis1_array=axis1_array,
+            axis2_array=axis2_array,
+            nside=nside,
+            healpix_pixel_array=healpix_pixel_array,
+            ordering=ordering,
+            history=f"created from a pyuvdata analytic beam: {self.__repr__()}",
+        )
+
+        if uvb.pixel_coordinate_system == "healpix":
+            try:
+                from astropy_healpix import HEALPix
+            except ImportError as e:
+                raise ImportError(
+                    "astropy_healpix is not installed but is "
+                    "required for healpix functionality. "
+                    "Install 'astropy-healpix' using conda or pip."
+                ) from e
+            hp_obj = HEALPix(nside=uvb.nside, order=uvb.ordering)
+            hpx_lon, hpx_lat = hp_obj.healpix_to_lonlat(uvb.pixel_array)
+            za_array, az_array = utils.coordinates.hpx_latlon_to_zenithangle_azimuth(
+                hpx_lat.radian, hpx_lon.radian
+            )
+
+        else:
+            az_array, za_array = np.meshgrid(uvb.axis1_array, uvb.axis2_array)
+            az_array = az_array.flatten()
+            za_array = za_array.flatten()
+
+        if beam_type == "efield":
+            eval_function = "efield_eval"
+        else:
+            eval_function = "power_eval"
+
+        data_array = getattr(self, eval_function)(
+            az_array=az_array, za_array=za_array, freq_array=freq_array
+        )
+
+        if uvb.pixel_coordinate_system == "az_za":
+            data_array = data_array.reshape(uvb.data_array.shape)
+
+        uvb.data_array = data_array
+
+        uvb.check()
+        return uvb
+
+
+def analytic_beam_constructor(loader, node):
+    """
+    Define a yaml constructor for analytic beams.
+
+    The yaml must specify a "class" field with an importable class and any
+    required inputs to that class's constructor.
+
+    Parameters
+    ----------
+    loader: yaml.Loader
+        An instance of a yaml Loader object.
+    node: yaml.Node
+        A yaml node object.
+
+    Returns
+    -------
+    beam
+        An instance of an AnalyticBeam subclass.
+
+    """
+    values = loader.construct_mapping(node)
+    if "class" not in values:
+        raise ValueError("yaml entries for AnalyticBeam must specify a class")
+    class_parts = (values.pop("class")).split(".")
+    class_name = class_parts[-1]
+    if len(class_parts) == 1:
+        # no module specified, assume pyuvdata
+        module = importlib.import_module("pyuvdata")
+    else:
+        module = (".").join(class_parts[:-1])
+        module = importlib.import_module(module)
+    beam_class = getattr(module, class_name)
+
+    beam = beam_class(**values)
+
+    return beam
+
+
+yaml.add_constructor("!AnalyticBeam", analytic_beam_constructor, Loader=yaml.SafeLoader)
```

Comment:
I would _also_ add it to the `FullLoader` since people sometimes use that (even if pyuvsim doesn't)